playbooks/group_vars/prometheus/public.yml

```diff
@@ -25,6 +25,124 @@ prometheus_storage__tsdb__retention: 30d
+      # https://www.robustperception.io/reduce-noise-from-disk-space-alerts/
+      expr: (100 * (1 - (predict_linear(node_filesystem_free{fstype="ext4"}[1h], 4 * 3600)/node_filesystem_size{fstype="ext4"}))) > {{ PROMETHEUS_ALERT_DISK_SPACE_THRESHOLD }}
+      for: '5m'
+      labels:
+        severity: warning
+      annotations:
+        summary: 'Instance [[ $labels.instance ]] disk is filling up.'
+        description: 'Instance disk may fill up {{ PROMETHEUS_ALERT_DISK_SPACE_THRESHOLD }}% in 4 hours.'
+  - name: HAProxyRules
+    rules:
+    - alert: haproxy_down
+      expr: avg_over_time(haproxy_up [5m]) < 0.91
+      for: '10s'
+      labels:
+        severity: critical
+      annotations:
+        summary: 'HAProxy [[ $labels.instance ]] down.'
+        description: '[[ $labels.instance ]] of job [[ $labels.job ]] is down.'
+    - alert: haproxy_backend_response_time_high
+      expr: haproxy_backend_http_response_time_average_seconds{node='{{ PROMETHEUS_ALERT_HAPROXY_NODES_REGEX }}'} > PROMETHEUS_ALERT_HTTP_RESPONSE_TIME_THRESHOLD
+      for: '5m'
+      labels:
+        severity: warning
+      annotations:
+        summary: 'Instance [[ $labels.instance ]] http response time high.'
+        description: 'HTTP response time of [[ $labels.instance ]] of job [[ $labels.job ]] has been above {{ PROMETHEUS_ALERT_HTTP_RESPONSE_TIME_THRESHOLD }} for 5 minutes.'
+    - alert: haproxy_frontend_request_errors_high
+      expr: (100 * (haproxy_frontend_request_errors_total{node='{{ PROMETHEUS_ALERT_HAPROXY_NODES_REGEX }}'}/haproxy_frontend_http_requests_total)) > PROMETHEUS_ALERT_HTTP_RESPONSE_FRONTEND_ERROR_THRESHOLD
+      for: '5m'
+      labels:
+        severity: warning
+      annotations:
+        summary: 'Instance [[ $labels.instance ]] http response time high.'
+        description: 'HTTP response time of [[ $labels.instance ]] of job [[ $labels.job ]] has been above {{ PROMETHEUS_ALERT_HTTP_RESPONSE_TIME_THRESHOLD }} for 5 minutes.'
+  - name: MySQLRules
+    rules:
+    - alert: mysql_down
+      expr: avg_over_time(mysql_up[5m]) < 0.91
+      for: '10s'
+      labels:
+        severity: critical
+      annotations:
+        summary: 'MySQL instance [[ $labels.instance ]] down.'
+        description: '[[ $labels.instance ]] of job [[ $labels.job ]] is down.'
+
+# ALERT MANAGER ################################################################
+
+prometheus_alert_manager_web__listen_address: '127.0.0.1:9093'
+
+prometheus_alert_manager_data__retention: '300h0m0s'
+
+prometheus_alert_manager_config_receivers:
+  - name: 'default'
+    email_configs:
+    - send_resolved: true
+      to: '{{ PROMETHEUS_ALERT_MANAGER_SMTP_TO }}'
+  - name: 'pager'
+    pagerduty_configs:
+    - send_resolved: true
+      service_key: '{{ PROMETHEUS_ALERT_MANAGER_PAGERDUTY_SERVICE_KEY }}'
+
+prometheus_alert_manager_config_route:
+  receiver: 'default'
+  repeat_interval: 4h
+  routes:
+    - receiver: 'pager'
+      repeat_interval: 5m
+      match:
```

Comment:
According to [the docs](https://prometheus.io/docs/alerting/configuration/#route), I get the impression that _all_ `match` labels need to be available in the alert, but we actually aren't sending the `environment` label for any rule, so this may not match anything and thus never page.

Reply:
Hmm. In that case, I can add the `environment` to the alert.